handbook/repository/workspace-settings.md

```diff
@@ -0,0 +1,180 @@
+# Workspace Settings
+
+Various project settings can be controlled from the `workspace.json`
+file, located in the root of the repository.
+
+## E2E Testing configuration for timeouts
+
+We are experiencing timeouts during some of our `Cypress` end-to-end
+tests. This is due to long build and compile time of `NextJS` apps.
+So we have a seperate `e2e-ci` task setup in the `workspace.json`
+configuration of each `e2e` app. With this configuration we can let
+`Cypress` wait until the server is ready before starting it's test
+execution.
+
+### NextJS apps pre-requisites
+
+#### Setting API_MOCKS
+
+We need to make sure that the `API_MOCKS` environment variable is set in
+the `next.config.js`
+
+```javascript
+// ...
+serverRuntimeConfig: {
+  // ...
+},
+publicRuntimeConfig: {
+  // ...
+},
+env: {
+  API_MOCKS: process.env.API_MOCKS,
+},
+```
+
+### React apps pre-requisites
+
+For React apps we first need to add a task `static-serve`
+in the target web app. For example for the `service-portal-e2e`
+it is `service-portal`
+
+```json
+"static-serve": {
+  "builder": "@nrwl/workspace:run-commands",
+  "options": {
+    "commands": [
+      {
+        "command": "yarn serve -l 4200 -s dist/apps/service-portal",
+        "forwardAllArgs": false
+      }
+    ],
+    "readyWhen": "Accepting connections"
+  }
+},
+```
+
+Here you need to change the path that follows the `-s`
+parameter for the corresponding `dist` folder of your
+built app.
+
+_Additionally_ you can change the `port` parameter by
+changing the value that comes after `-l` option.
+
+> :information*source: \_React* apps need additional cleanup
+> in their `e2e-ci` task. See [e2e-ci for React](#React)
+
+Next we change the `devServerTarget` for the `production` config of
+the `e2e` task in the `e2e` project to use this new `static-serve` task.
+
+```json
+"e2e": {
+  ...
+  "configurations": {
+    "production": {
+      "devServerTarget": "service-portal:static-serve"
+    }
+  }
+},
+```
+
+#### Setting API_MOCKS
+
+For React apps we have a common webpack config in `libs/share/webpack`
+which we use to make sure the `API_MOCKS` environment variable is set,
+along with other common plugins and settings.
+
+Your `webpack.config.js` should look like this:
+
+```javascript
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const nrwlConfig = require('./../../libs/shared/webpack/nrwl-config')
+
+module.exports = (config, context) => {
+  // Add our common webpack config
+  nrwlConfig(config)
+
+  // Here you can add app specific config
+
+  return {
+    ...config,
+    node: {
+      process: true,
+      global: true,
+    },
+  }
```

Comment:
I would just return the config, the node stuff may confuse the reader if it's not explained.